examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceTest.java

```diff
@@ -0,0 +1,73 @@
+package com.example.ollamahf;
+
+import io.restassured.http.Header;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.ContainerFetchException;
+import org.testcontainers.ollama.OllamaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class OllamaHuggingFaceTest {
+
+    @Test
+    public void embeddingModelWithHuggingFace() {
+        String repository = "CompendiumLabs/bge-small-en-v1.5-gguf";
+        String model = "bge-small-en-v1.5-q4_k_m.gguf";
+        String imageName = "embedding-model-from-hf";
+        try (
+            OllamaContainer ollama = new OllamaContainer(
+                DockerImageName.parse(imageName).asCompatibleSubstituteFor("ollama/ollama:0.1.42")
+            )
+        ) {
+            try {
+                ollama.start();
+            } catch (ContainerFetchException ex) {
+                // Create the image
+                try (
+                    OllamaHuggingFaceContainer huggingFaceContainer = new OllamaHuggingFaceContainer(
+                        imageName,
+                        new OllamaHuggingFaceContainer.HuggingFaceModel(repository, model)
+                    )
+                ) {
+                    huggingFaceContainer.start();
+                    huggingFaceContainer.stop();
+                }
+                ollama.start();
+            }
```

Comment:
let's use the docker client API to check if the image exists, the code will be more readable.